pkg/cmd/cli/restore/create.go

```diff
@@ -170,12 +179,65 @@ func (o *CreateOptions) Validate(c *cobra.Command, args []string, f client.Facto
 	return nil
 }
 
+// mostRecentBackup returns the backup with the most recent start timestamp that has a phase that's
+// in the provided list of allowed phases.
+func mostRecentBackup(backups []api.Backup, allowedPhases ...api.BackupPhase) *api.Backup {
+	// sort the backups in descending order of start time (i.e. most recent to least recent)
+	sort.Slice(backups, func(i, j int) bool {
+		// Use .After() because we want descending sort.
+
+		var iStartTime, jStartTime time.Time
+		if backups[i].Status.StartTimestamp != nil {
+			iStartTime = backups[i].Status.StartTimestamp.Time
+		}
+		if backups[j].Status.StartTimestamp != nil {
+			jStartTime = backups[j].Status.StartTimestamp.Time
+		}
+		return iStartTime.After(jStartTime)
+	})
+
+	// create a map of the allowed phases for easy lookup below
+	phases := map[api.BackupPhase]struct{}{}
+	for _, phase := range allowedPhases {
+		phases[phase] = struct{}{}
+	}
+
+	var res *api.Backup
+	for i, backup := range backups {
+		// if the backup's phase is one of the allowable ones, record
+		// the backup and break the loop so we can return it
+		if _, ok := phases[backup.Status.Phase]; ok {
+			res = &backups[i]
+			break
+		}
+	}
+
+	return res
+}
+
 func (o *CreateOptions) Run(c *cobra.Command, f client.Factory) error {
 	if o.client == nil {
 		// This should never happen
 		return errors.New("Velero client is not set; unable to proceed")
 	}
 
+	// if --allow-partially-failed was specified, look up the most recent Completed or
+	// PartiallyFailed backup for the provided schedule, and use that specific backup
+	// to restore from.
+	if o.ScheduleName != "" && boolptr.IsSetToTrue(o.AllowPartiallyFailed.Value) {
+		backups, err := o.client.VeleroV1().Backups(f.Namespace()).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", api.ScheduleNameLabel, o.ScheduleName)})
+		if err != nil {
+			return err
+		}
+
+		// if we find a Completed or PartiallyFailed backup for the schedule, restore specifically from that backup. If we don't
+		// find one, proceed as-is -- the Velero server will handle validation.
+		if backup := mostRecentBackup(backups.Items, api.BackupPhaseCompleted, api.BackupPhasePartiallyFailed); backup != nil {
+			o.BackupName = backup.Name
+			o.ScheduleName = ""
```

Comment:
Maybe we should add a TODO here so we know to come back and address this hack?

Reply:
good point, TODO added